### DISCUSSION TITLE
How can I export all _non-default_ settings?

### DISCUSSION THREAD

**Discussion by j-steinbach** [tridactyl/tridactyl#4246] [General]
I changed a few settings and binds and now I want to put them into a tridactylrc. I found the `mktridactylrc --clipboard` command, but this appears to export _all_ settings. How do I only get the settings that have changed, i.e. the _non-default_ settings?

**Comment by bovine3dom** [comment]
Hi, that shouldn't happen. I suspect somehow the entire default configuration has been copied somehow into your configuration at some point. If `:viewconfig --user` shows mostly default configuration that is what has happened.

Fixing that shouldn't be impossible - you could export and calculate the difference in the JSON between `:viewconfig --{user,default}` and then follow the guide on the wiki for migration between beta and stable to reimport it, or you could use `:js` with the `tri.config.{USERCONFIG,default_config()` objects and calculate the difference with maybe ramda (available as tri.R).

Good luck and ping me if you get stuck :)

**Reply by j-steinbach** [reply]
Your idea with comparing `user` and `default` did the trick, thanks! Comparing those, it seems that I mainly got exported a default copy of the nmap. Maybe because I mostly did changes there?.  I will now put it into a `tridactylrc` and try my luck with `native`.

----

Oh, but maybe you have the answer to another small question I have: How do I prevent the new-tab from automatically putting focus to the address bar? I want to open a new-tab and then utilize a command, but instead it writes my keystrokes into the address bar. I currently remove focus with `S-F6` every time, but that is getting a bit annoying.

**Reply by bovine3dom** [reply]
ctrl-t puts focus in the address bar but `:tabopen` does not. If you just press `t<Enter>` you'll find that focus is where you want it.

But if you're doing that you might as well use the `:tabopen` completions from the page you were already on :)

**Reply by j-steinbach** [reply]
Ok, so I guess it is just a matter of muscle-memory on my side :) (and I just maybe have to deal with it for the first default tab whenever I start Firefox). Thanks!!

**Reply by bovine3dom** [reply]
If you run this https://github.com/glacambre/firefox-patches/issues/1#issuecomment-547301077 you could do `:bind <C-t> tabopen`, but you would need to repatch every time Firefox updated.

You can get around "focus on the first tab" by making Firefox load a page, e.g. start it with `firefox news.bbc.co.uk` rather than `firefox`

**Reply by j-steinbach** [reply]
Doing the patch thing seems as if it would just move the annoyance from one side to the other, so probably not the best solution for me.

But I really like the pragmatic approach of simply starting with a "working" page. Simple, no hassles and maintainable. Thanks³ a lot!
